src/commands/lambda/__tests__/flare.test.ts

```diff
@@ -204,111 +228,405 @@ describe('lambda flare', () => {
     })
   })
 
-  describe('writeFile', () => {
-    const MOCK_DATA = 'mock data'
-    const context = createMockContext() as BaseContext
-    ;(fs.existsSync as jest.Mock).mockReturnValue(false)
+  describe('deleteFolder', () => {
+    it('successfully deletes a folder', async () => {
+      deleteFolder(MOCK_FOLDER_PATH)
 
-    it('successfully writes data to a file with no error', async () => {
-      await writeFile(MOCK_FOLDER_PATH, MOCK_FILE_PATH, MOCK_DATA, context)
+      expect(fs.rmSync).toHaveBeenCalledWith(MOCK_FOLDER_PATH, {recursive: true, force: true})
+    })
+
+    it('throws error when unable to delete a folder', async () => {
+      ;(fs.rmSync as jest.Mock).mockImplementation(() => {
+        throw new Error('MOCK ERROR: Unable to delete folder')
+      })
+
+      expect(() => deleteFolder(MOCK_FOLDER_PATH)).toThrowErrorMatchingSnapshot()
+      expect(fs.rmSync).toHaveBeenCalledWith(MOCK_FOLDER_PATH, {recursive: true, force: true})
+      ;(fs.rmSync as jest.Mock).mockRestore()
+    })
+  })
+
+  describe('createDirectories', () => {
+    const MOCK_LOG_PATH = path.join(MOCK_FOLDER_PATH, 'logs')
+    it('successfully creates a root folder', async () => {
+      createDirectories(MOCK_FOLDER_PATH, MOCK_LOG_PATH, new Map())
+
+      expect(fs.mkdirSync).toHaveBeenCalledWith(MOCK_FOLDER_PATH)
+    })
+
+    it('successfully creates a root and logs folder', async () => {
+      createDirectories(MOCK_FOLDER_PATH, MOCK_LOG_PATH, MOCK_LOGS)
 
-      expect(fs.existsSync).toHaveBeenCalledWith(MOCK_FOLDER_PATH)
       expect(fs.mkdirSync).toHaveBeenCalledWith(MOCK_FOLDER_PATH)
-      expect(fs.writeFileSync).toHaveBeenCalledWith(MOCK_FILE_PATH, MOCK_DATA)
+      expect(fs.mkdirSync).toHaveBeenCalledWith(MOCK_LOG_PATH)
     })
 
-    it('throws error when unable to create folder', async () => {
+    it('throws error when unable to create a folder', async () => {
       ;(fs.mkdirSync as jest.Mock).mockImplementation(() => {
         throw new Error('MOCK ERROR: Unable to create folder')
       })
 
-      await expect(writeFile(MOCK_FOLDER_PATH, MOCK_FILE_PATH, MOCK_DATA, context)).rejects.toMatchSnapshot()
-
-      expect(fs.existsSync).toHaveBeenCalledWith(MOCK_FOLDER_PATH)
+      expect(() => createDirectories(MOCK_FOLDER_PATH, MOCK_LOG_PATH, new Map())).toThrowErrorMatchingSnapshot()
       expect(fs.mkdirSync).toHaveBeenCalledWith(MOCK_FOLDER_PATH)
-      ;(fs.mkdirSync as jest.Mock).mockRestore()
+      fs.mkdirSync = jest.fn().mockImplementation(() => {})
+    })
+  })
+
+  describe('getLogStreamNames', () => {
```

Comment:
This tests are not using the AWS SDK mocking library, there should be examples about how client commands are mocked in the repo.

Let's change this tests to use the same mocking strategy for the AWS SDK.

Reply:
Changed my `getLogStreamNames` and `getLogEvents` tests, can you take another look?